docs/spec_draft.md

```diff
@@ -1061,6 +1099,74 @@ tensor and produces a `result` tensor.
+  * `has_side_effect` is true if the custom call has side-effects.
+  * `backend_config` can be used to encode additional information.
+  * `api_version` specifies the version of the custom call API.
+  * `called_computations` used to apply functions within the scope of the parent
+    module.
+  * `operand_layouts` specifies the layout of the operand.
+  * `result_layouts` specifies the layout of the result.
+  * `output_operand_aliases` asiases attributes for results and operands.
+
+### Inputs
+
+| Name                     | Type                                                                |
+|--------------------------|---------------------------------------------------------------------|
+| `inputs`                 | variadic number of tensors of any supported type, tokens, or tuples |
+| `call_target_name`       | `string`                                                            |
+| `has_side_effect`        | `i1`                                                                |
+| `backend_config`         | `string`                                                            |
+| `api_version`            | enum of `API_VERSION_ORIGINAL`, `API_VERSION_STATUS_RETURNING`,     |
+|                          | and `API_VERSION_STATUS_RETURNING_UNIFIED`                          |
+| `called_computations`    | array of `function` references                                      |
+| `operand_layouts`        | array of 1-dimensional tensor constants of type `index`             |
+| `result_layouts`         | array of 1-dimensional tensor constants of type `index`             |
+| `output_operand_aliases` | [`OutputOperandAlias`](#stablehlooutput_operand_alias)              |
+
+### Outputs
+
+| Name      | Type                                                                |
+|-----------|---------------------------------------------------------------------|
+| `results` | variadic number of tensors of any supported type, tokens, or tuples |
+
+### Constraints
+
+  * (C1) `operand_layouts` & `result_layouts` attributes can be specified under
+          the following constraints:
+    * Either both `operand_layouts` and `result_layouts` are specified or none.
+    * None of the `inputs` are of tuple type.
+    * None of the `results` are of tuple type except the common case of single
+      tuple result packing non-tuple values is allowed.
+    * `operand_layouts[i]` = `layout(inputs[i])`, for all `i`.
+    * `result_layouts[i]` = `layout(results[i])`, for all `i`.
+  * (C2) `output_operand_aliases` can be specified under the following
+         constraints:
+    * `type(inputs[operand_index])` = `tuple`.
+    * `shape(inputs[operand_index][operand_tuple_indices[0]]..[operand_tuple_indices[N-1]])` =
+      `shape(results[output_tuple_indices[0]]..[output_tuple_indices[M-1]])`,
+      where `size(operand_tuple_indices) = N`, `size(output_tuple_indices) = M`.
+
+### Examples
+
+```mlir
+%results:3 = "stablehlo.custom_call"(%inputs0, %inputs1) {
```

Comment:
https://github.com/openxla/stablehlo/blob/main/docs/spec_draft.md#examples-49
for outfeed https://github.com/openxla/stablehlo/blob/main/docs/spec_draft.md#examples-49 inputs is not defined.
I think it is because we cant demonstrate how the results depend on the inputs, the values of the inputs are not useful.

Reply:
are you talking about input parameters?

Reply:
Like this?

```
%results = "stablehlo.custom_call"(%inputs0) {
    call_target_name = "foo",
    has_side_effect = false,
    backend_config = "",
    api_version = 1 : i32,
    called_computations = [@foo],
    operand_layouts = [dense<> : tensor<0xindex>],
    output_operand_aliases = [
      #stablehlo.output_operand_alias<output_tuple_indices = [],
                                 operand_index = 0,
                                 operand_tuple_indices = []>],
    result_layouts = [dense<> : tensor<0xindex>]
  } : (tensor<f32>) -> tensor<f32>
  func.return %0 : tensor<f32>
}
```